Move the autoscaling e2e tests to be ran in parallel

kamarabbas99 opened the pull request:
#### What type of PR is this?
This PR gives an opportunity to us to run these e2e tests in parallel since we have no coverage for HPAs being processed concurrently (concurrent processing of HPAs was added in [1cefcde](https://github.com/kubernetes/kubernetes/commit/1cefcdea2d6195a794ef122cb7a36614a580d41b)). Also these test were moved to serial because of [#20338](https://github.com/kubernetes/kubernetes/issues/20338) but that issue is now closed.
<!--
Add one of the following kinds:
/kind feature
-->

#### What this PR does / why we need it:
We will have test coverage for multiple HPAs being processed concurrently([1cefcde](https://github.com/kubernetes/kubernetes/commit/1cefcdea2d6195a794ef122cb7a36614a580d41b)).
#### Which issue(s) this PR fixes:
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

kamarabbas99 committed:
Move the autoscaling e2e tests to be ran in parallel to have coverage for concurrent hpa processing

kamarabbas99 committed:
Move the autoscaling e2e tests to be ran in parallel to have coverage for concurrent hpa processing

raywainman commented:
/ok-to-test

/lgtm

omerap12 commented on test/e2e/framework/autoscaling/autoscaling_utils.go:
```diff
@@ -61,7 +61,7 @@ const (
 	targetPort                      = 8080
 	sidecarTargetPort               = 8081
-	timeoutRC                       = 120 * time.Second
+	timeoutRC                       = 5 * time.Minute
```
Why are we increasing the timeout from 120 seconds to 5 minutes if the tests run in parallel? Shouldn't they be faster, not slower?

omerap12 reviewed (commented):
Thanks for this :) 
Just one comment from me

kamarabbas99 committed:
Update autoscaling_utils.go

kamarabbas99 reviewed (commented):
Addressed the comment! So now I am only removing the tags from these specs. To run it in parallel we still need to use gingkgo cli. These chnages alone wont run specs in parallel.

omerap12 commented:
> Addressed the comment! So now I am only removing the tags from these specs. To run it in parallel we still need to use gingkgo cli. These chnages alone wont run specs in parallel.

Yes. this needed change in test-infra as well.

omerap12 commented:
/lgtm

kamarabbas99 commented:
/retest-required

soltysh commented:
/label tide/merge-method-squash

soltysh reviewed (approved):
/approve

kamarabbas99 commented:
> > Addressed the comment! So now I am only removing the tags from these specs. To run it in parallel we still need to use gingkgo cli. These chnages alone wont run specs in parallel.
> 
> Yes. this needed change in test-infra as well.

has that change already been merged? I dont think these tests are currently running in parallel, https://testgrid.k8s.io/sig-autoscaling-hpa#gci-gce-autoscaling-hpa-cpu-pull
